### PR TITLE
JmxBuilder export bean:specifying MBean ObjectName support GString.

### PR DESCRIPTION
**Pull request by hdfg159**
https://issues.apache.org/jira/browse/GROOVY-9829

### CODE REVIEW

File: subprojects/groovy-jmx/src/main/groovy/groovy/jmx/builder/JmxMetaMapBuilder.groovy [hunk 1]
```diff
@@ -172,7 +172,7 @@ class JmxMetaMapBuilder {
     private static ObjectName getObjectName(def map) {
         if (!map) return null
         def jmxName
-        if (map.name instanceof String) {
+        if (map.name instanceof String || map.name instanceof GString) {
```
Comment (eric-milles) [inline subprojects/groovy-jmx/src/main/groovy/groovy/jmx/builder/JmxMetaMapBuilder.groovy:175]: Would "instanceof CharSequence" cover it?

Comment (paulk-asert) [reply subprojects/groovy-jmx/src/main/groovy/groovy/jmx/builder/JmxMetaMapBuilder.groovy:175]: I'll merge but tweak with the suggestion from @eric-milles

Comment (hdfg159) [reply subprojects/groovy-jmx/src/main/groovy/groovy/jmx/builder/JmxMetaMapBuilder.groovy]: Thanks for the suggestion.

**Comment by paulk-asert** [comment]
Merged, thanks!

**Pull request closed without merging**